toolkit/committee-selection/authority-selection-inherents/src/authority_selection_inputs.rs

```diff
@@ -34,21 +41,31 @@ pub enum AuthoritySelectionInputsCreationError {
 			"Failed to get registered candidates for epoch: {0}, committee candidate address: {1}: {2}."
 		)
 	)]
+	/// Failed to get registered candidates for epoch
 	GetCandidatesQuery(McEpochNumber, String, Box<dyn std::error::Error + Send + Sync>),
 	#[cfg_attr(feature = "std", error("Failed to get epoch nonce for epoch: {0}: {1}."))]
+	/// Failed to get epoch nonce for epoch
 	GetEpochNonceQuery(McEpochNumber, Box<dyn std::error::Error + Send + Sync>),
 }
 
 #[derive(Debug, Clone, PartialEq, serde::Serialize)]
+/// Permissioned candidate data from Cardano main chain
 pub struct RawPermissionedCandidateData {
+	/// Unverified Partner Chain public key of permissioned candidate
```

Comment:
I would rather go with "validation". If parsing these bytes into an ECDSA or EDDSA or other proper cryptographic key fails, you would rather receive "invalid key" rather then "unable to verify" error.